soc/ti/k3/am6x/m4/soc.c

```diff
@@ -44,30 +52,88 @@ static struct address_trans_region_config region_config[] = {
  */
 };
 
-static void mmr_unlock(void)
+static void am62x_mmr_unlock(void)
 {
-	uint32_t baseAddr = PINCTRL_BASE_ADDR;
+	uint32_t baseAddr = AM62_PINCTRL_BASE_ADDR;
 	uintptr_t kickAddr;
 
 	/* Lock 0 */
-	kickAddr = baseAddr + CSL_MCU_PADCONFIG_LOCK0_KICK0_OFFSET;
-	sys_write32(KICK0_UNLOCK_VAL, kickAddr);   /* KICK 0 */
+	kickAddr = baseAddr + AM62_CSL_MCU_PADCONFIG_LOCK0_KICK0_OFFSET;
+	sys_write32(AM62_KICK0_UNLOCK_VAL, kickAddr);   /* KICK 0 */
 	kickAddr = kickAddr + sizeof(uint32_t *);
-	sys_write32(KICK1_UNLOCK_VAL, kickAddr);   /* KICK 1 */
+	sys_write32(AM62_KICK1_UNLOCK_VAL, kickAddr);   /* KICK 1 */
 
 	/* Lock 1 */
-	kickAddr = baseAddr + CSL_MCU_PADCONFIG_LOCK1_KICK0_OFFSET;
-	sys_write32(KICK0_UNLOCK_VAL, kickAddr);   /* KICK 0 */
+	kickAddr = baseAddr + AM62_CSL_MCU_PADCONFIG_LOCK1_KICK0_OFFSET;
+	sys_write32(AM62_KICK0_UNLOCK_VAL, kickAddr);   /* KICK 0 */
 	kickAddr = kickAddr + sizeof(uint32_t *);
-	sys_write32(KICK1_UNLOCK_VAL, kickAddr);   /* KICK 1 */
+	sys_write32(AM62_KICK1_UNLOCK_VAL, kickAddr);   /* KICK 1 */
 }
 
 static int am62x_m4_init(void)
 {
-	sys_mm_drv_ti_rat_init(
-		region_config, ADDR_TRANSLATE_RAT_BASE_ADDR, ARRAY_SIZE(region_config));
-	mmr_unlock();
+	sys_mm_drv_ti_rat_init(am62x_region_config, AM62_ADDR_TRANSLATE_RAT_BASE_ADDR,
+		ARRAY_SIZE(am62x_region_config));
+	am62x_mmr_unlock();
 	return 0;
 }
 
 SYS_INIT(am62x_m4_init, EARLY, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
+#endif
+
+#ifdef CONFIG_SOC_AM6442_M4
```

Comment:
If this is same for AM6234/AM6442, I'd just wrap in in `CONFIG_SOC_SERIES_AM6X_M4` rather than duplicate, there are lots of SoCs in this family

Reply:
@vaishnavachath what do you think? Maybe we should introduce SOC_AM62X_M4/SOC_AM64x_M4 configs? I don't think we need to define socs with the full part number like AM6234 here in Zephyr since we don't care about them.

Reply:
From a quick look on the TI site there is currently AM62/AM64/AM65/AM67/AM68/AM69 series.
IMHO creating duplication later is easy, removing it is difficult 🤷

Reply:
@gramsay0  @dnltz I checked the TI SDK offerings, the default configurations are same for AM64 and AM62 so it can be reused now without duplication. 
But for the rest of the devices AM67/AM68/AM69 .etc even though the devices are of same family with increasing capabilities, the memory map might be different and such reuse won't be possible, for example only some of the devices  offer a general purpose M4 core which is programmable by the user, the reuse and extension can apply to devices of same family - like AM62-AM62A-AM62P .etc

Another concern is that RAT configuration is use case dependent and a global RAT configuration might not be suitable for all use cases and in TI SDKs also the RAT configs are configurable per example/application. that  was one of the reasons suggesting duplicate configs so that when someone changes the default configs for one SoC family it should not cause silent failures in other SoC family, RAT config failures are a bit tricky to debug also since the driver does not expose any of the diagnostics features.

Reusing is good for now, but I will recommend atleast adding a comment mentioning that this configs are shared between multiple devices.